imports/ui/layouts/App.jsx

```diff
@@ -1,166 +1,94 @@
 
-  logout() {
+  const logout = () => {
     Meteor.logout();
-    this.setState({
-      redirectTo: this.state.defaultList,
-    });
-  }
+    setRedirectTo(defaultList);
+  };
 
-  renderRedirect(location) {
-    const { redirectTo, defaultList } = this.state;
+  const renderRedirect = (location) => {
     const { pathname } = location;
     let redirect = null;
     if (redirectTo && redirectTo !== pathname) {
       redirect = <Redirect to={redirectTo} />;
     } else if (pathname === '/' && defaultList) {
       redirect = <Redirect to={defaultList} />;
     }
+    setRedirectTo(null);
     return redirect;
-  }
+  };
 
-  renderContent(location) {
-    const {
-      user,
-      connected,
-      lists,
-      menuOpen,
-      loading,
-    } = this.props;
-    const { showConnectionIssue } = this.state;
+  renderRedirect.propTypes = {
+    pathname: PropTypes.string.isRequired,
+  };
 
-    const commonChildProps = {
-      menuOpen: this.props.menuOpen,
-    };
-
-    return (
-      <div id="container" className={menuOpen ? 'menu-open' : ''}>
-        <section id="menu">
-          <LanguageToggle />
-          <UserMenu user={user} logout={this.logout} />
-          <ListList lists={lists} />
-        </section>
-        {showConnectionIssue && !connected
-          ? <ConnectionNotification />
-          : null}
-        <div className="content-overlay" onClick={this.closeMenu} />
-        <div id="content-container">
-          {loading ? (
-            <Loading key="loading" />
-          ) : (
-            <TransitionGroup>
-              <CSSTransition
-                key={location.key}
-                classNames="fade"
-                timeout={200}
-              >
-                <Switch location={location}>
-                  <Route
-                    path="/lists/:id"
-                    render={({ match }) => (
-                      <ListPageContainer match={match} {...commonChildProps} />
-                    )}
-                  />
-                  <Route
-                    path="/signin"
-                    render={() => <AuthPageSignIn {...commonChildProps} />}
-                  />
-                  <Route
-                    path="/join"
-                    render={() => <AuthPageJoin {...commonChildProps} />}
-                  />
-                  <Route
-                    path="/*"
-                    render={() => <NotFoundPage {...commonChildProps} />}
-                  />
-                </Switch>
-              </CSSTransition>
-            </TransitionGroup>
-          )}
-        </div>
-      </div>
-    );
-  }
-
-  render() {
-    return (
+  return (
+    <GlobalStateProvider>
       <BrowserRouter>
         <Route
           render={({ location }) => (
-            this.renderRedirect(location) || this.renderContent(location)
+            renderRedirect(location) || (
+              <AppContent
+                connexionNotification={showConnectionIssue && !connected}
```

Comment:
This is probably a typo

Reply:
What do you mean? Does it sounds better to you this way?
```
render={({ location }) => renderRedirect(location) || (
  <AppContent
    connexionNotification={showConnectionIssue && !connected}
    lists={lists}
    loading={loading}
    location={location}
    logout={logout}
    user={user}
  />
)}
```
Anyway, once redirection will be handled by useHistory, that won't remain like that.

Reply:
Sorry, GitHub displayed more lines haha, I meant the `connexionNotification` property

Reply:
:D
It took me a while to get it! I didn't see the problem since... I'm French and in French, we do write it 'connexion'.